src/metrics/hooks/useFetchCpuUsageMetrics.tsx

```diff
@@ -29,6 +29,19 @@ export const useFetchCpuUsageMetrics = (
   return ({ name, namespace, span, samples, endTime, timeout, delay }) => {
     [...Array(n)].map((_, i) => {
       const query = cpuUsageQuery(name, namespace, i);
+      // Normally it is impossible to call a hook in a conditional branch (if,
+      // loops). That is because it would mess with how React stores it's data.
+      // https://react.dev/reference/rules/rules-of-hooks
+      //
+      // However in this specific configuration, the number `n` determining how
+      // many loops are going to occur gets defined by the number of pods within
+      // the deployment. This means this number is guaranteed to stay the same
+      // while the page is up and running, as long as there is no support for
+      // hot reloading.
+      //
+      // That said, it means that it is therefore safe to use this hook in this
+      // loop.
+      // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
@lavocatt as far i know disabling ESLint warnings with inline comments to fix issues is not the recommended approach in the codebase. Instead, we strive to maintain consistency and manage ESLint rules globally through our ESLint configuration file (eslintrc.yaml).

Reply:
yes, it's makes sense to me. and i agree with point.